db/migrate/20200213210949_add_insert_size_metrics_to_pipeline_runs.rb

```diff
@@ -0,0 +1,12 @@
+class AddInsertSizeMetricsToPipelineRuns < ActiveRecord::Migration[5.1]
+  def change
```

Comment:
Then you can add some rails validators as well to describe the expected data.

Reply:
I implemented this with a separate table (see https://github.com/chanzuckerberg/idseq-web/pull/3076/commits/7f082910bb4e070abcfb15aaf0727346b43ebdc1) and changed it back. In my opinion it made the code way more complex. It introduced nesting and I ended up needing to ensure the values were present in multiple contexts. While the ercc counts have their own table there are more than one per sample, which requires the secondary table. The total ercc count is in the `pipeline_runs` table. Regarding the historical nulls, these are optional fields so I don't think it's an issue that the historical runs have null values in them.

Reply:
Thanks for trying out the different structure. Although I still prefer a separate table, I acknowledge that it is mostly a philosophical question of data modeling. I would rather work with a separate table containing 8 not-null columns than 8 columns in `pipeline_runs` which are non-null when the host is human. See this post for a good summary of all the pros and cons: https://dba.stackexchange.com/questions/103855/is-it-better-to-create-another-table-to-possible-null-values-in-relational-datab . 

With that in mind, I would be happy to defer to someone else as a "tie-breaker" for this design question. Having some consensus in the team would be good if nothing else. 

Sidenote: you should estimate the time to migrate pipeline_runs in prod if we add the cols to that table, or confirm that it is non-blocking DDL. We've had problems with long migrations in the past. 

PS: For encapsulation, another possible design is to put all the insert size columns in a JSON object.

Reply:
Most precedents so far are on adding the extra columns on the `pipeline_runs` tables. 
Not sure if ERCCs are an equivalent example since it is a 1:N relation vs 1:1.
Having said that, I agree that this is a good example to flag. This is not typically the way you add statistics to a database and is just not sustainable to add 8 fields of statistic to a core table like pipeline runs. 
(Regarding JSON, we need to be careful when using it. There is not good support for json in our database, in particular for querying.)
Questions about the data, what makes insert size metrics different? Why are we adding 8 columns vs 1? If multiple statistics are required how do we know we not going to need more in the future (in which case we should store the raw values somewhere? Are we fully sure that this are the relevant statistics and they are not going to be added/edited/remove later on?
I tend to agree with @gregdingle that the best solution seems to be to put them on a separate table. Code could be slightly more complex due to the extra join, but i do not think it would affect much (@morsecodist can you give an example of worst case code complexity?). Also, as we migrate the pipeline, we might factor data out, and a separate table would make those tasks easier.

Reply:
You've convinced me to move back to the multi-table. The code complexity was mostly an issue of ensuring we had the sample details on the front end. 

What do you mean what makes them different? I am saving more values than we explicitly use because I feel they add some additional insight and we may want them. The raw files are stored in s3 so we could always access them again. The nice thing about a separate table is I can make them all required since we either have all or none.